File: Backend/Battle_System/Card.py
class Card:
    def __init__(self, name, health, damage, ability):
        self.name = name
        self.max_health = health
        self.base_health = health
        self.health = health
        self.damage = damage
        self.ability = ability
        self.block = 0
        self.temp_effects = []

    def attack(self, target, ap):
        if self.health <= 0:
            print(f"{self.name} is defeated and cannot attack!") 
            return 0  # Dead cards can't attack

        if ap <= 0:
            return 0
        
        base_damage = self.damage
        #if "damage_multiplier" in self.temp_effects:
            #base_damage = int(base_damage * 1.5)
        
        #if ap == 2:
            #base_damage = int(base_damage * 1.5)
        
        actual_damage = min(target.health, base_damage)
        target.health -= actual_damage

        if target.health <= 0:
            target.health = 0
        
        return actual_damage
    
    def use_ability(self, player):
        if self.ability == "heal":
            heal_amount = 75
            self.health = min(self.max_health, self.health + heal_amount)
            return f"{self.name} heals for {heal_amount} HP!"
        elif self.ability == "block":
            self.block = 80
            self.health += self.block
            return f"{self.name} gains {self.block} temporary HP!"
        #elif self.ability == "damage_multiplier":
            #self.temp_effects.append("damage_multiplier")
            #return f"{self.name} gets a damage boost!"
        elif self.ability == "free_switch":
            player.switch_card(free=True)
            return f"{player.name} switches cards for free!"
        return "Invalid ability"
    
    def remove_block(self):
        if self.block > 0:
            self.health -= self.block
            self.block = 0
            if self.health < self.base_health:
                self.health = self.base_health

    def __repr__(self):
        return f"{self.name} (HP: {self.health}/{self.max_health}, DMG: {self.damage})"

File: Backend/Battle_System/Card.js
class Card {
    constructor(name, health, damage, ability) {
      this.name = name;
      this.maxHealth = health;
      this.baseHealth = health;
      this.health = health;
      this.damage = damage;
      this.ability = ability;
      this.block = 0;
      this.tempEffects = [];
    }
  
    attack(target, ap) {
      if (this.health <= 0) {
        console.log(`${this.name} is defeated and cannot attack!`);
        return 0;  // Dead cards can't attack
      }
  
      if (ap <= 0) {
        return 0;
      }
  
      let baseDamage = this.damage;
      const actualDamage = Math.min(target.health, baseDamage);
      target.health -= actualDamage;
  
      if (target.health <= 0) {
        target.health = 0;
      }
  
      return actualDamage;
    }
  
    useAbility(player) {
      if (this.ability === "heal") {
        const healAmount = 75;
        this.health = Math.min(this.maxHealth, this.health + healAmount);
        return `${this.name} heals for ${healAmount} HP!`;
      } else if (this.ability === "block") {
        this.block = 80;
        this.health += this.block;
        return `${this.name} gains ${this.block} temporary HP!`;
      } else if (this.ability === "free_switch") {
        player.switchCard(true);
        return `${player.name} switches cards for free!`;
      }
      return "Invalid ability";
    }
  
    removeBlock() {
      if (this.block > 0) {
        this.health -= this.block;
        this.block = 0;
        if (this.health < this.baseHealth) {
          this.health = this.baseHealth;
        }
      }
    }
  
    toString() {
      return `${this.name} (HP: ${this.health}/${this.maxHealth}, DMG: ${this.damage})`;
    }
  }
  